Backpack-SwiftUI/FieldSet/Classes/BPKFieldSet.swift

```diff
@@ -0,0 +1,195 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2023 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import UIKit
+
+public protocol BPKFieldSetStatusHandling: View {
+    associatedtype WrappedView: BPKFieldSetStatusHandling
+    
+    func inputState(_ state: BPKFieldSet<WrappedView>.State) -> WrappedView
+}
+
+// swiftlint:disable line_length
+
+/// A component which wraps its content (view) and optionally adds a title, description and error label (depending on the field's state) around it.
+/// Supported states are Default, and Error. The states are dispatched to the wrapped view. The wrapped view must conform to `BPKFieldSetStatusHandling` to ensure it can handle the dispatched state.
+///
+/// Use `inputState(_ state: State)` to change the state of the field set.
+///
+/// Use `accessibilityPrefix(_ prefix: String)` to add a prefix to each accessibilityIdentifier
+/// that gets added to all fieldset's subcomponents
+
+// swiftlint:enable line_length
+
+public struct BPKFieldSet<Content: BPKFieldSetStatusHandling>: View {
+    private var state: BPKFieldSet<Content.WrappedView>.State = .default
+    private let label: String?
+    private let content: Content
+    private let description: String?
+    private var accessibilityPrefix: String?
+    
+    public init(
+        label: String? = nil,
+        description: String? = nil,
+        content: () -> Content
+    ) {
+        self.label = label
+        self.description = description
+        self.content = content()
+    }
+    
+    public var body: some View {
+        VStack(alignment: .leading, spacing: .sm) {
+            labelView
+            content
+                .inputState(state)
```

Comment:
I'm not quite sure I understand how this works? the implementations in the Text field, text area and BPKSelect just return mapping a state? but how is the style applied in the end I dont understand

Reply:
Just got it! it's calling each element's method!